ci/tests/vmachine/vmachine_test.py

```diff
@@ -0,0 +1,124 @@
+    if 'not' not in out:
+        general.execute_command('rm -rf {0}'.format(template_target_folder))
+        general.execute_command('mkdir {0}'.format(template_target_folder))
+    download_template(get_template_location_by_ip(GRID_IP))
+
+
+def setup():
+    check_template_exists()
+    vpool_test.setup()
+    mgmt_center_test.setup()
+    vpool_test.add_vpool()
+
+
+def teardown():
+    api = Connection.get_connection()
+    vpool_list = api.get_component_by_name('vpools', VPOOL_NAME)
+    assert len(vpool_list), "No vpool found where one was expected"
+    vpool = vpool_list[0]
+    if vpool:
+        _, __ = general.execute_command("rm -rf /mnt/{0}/*.raw".format(VPOOL_NAME))
+        general.api_remove_vpool(VPOOL_NAME)
+    vpool_test.teardown()
+    # @todo change mgmgt teardown to delete recently added hmc
+    mgmt_center_test.teardown()
+
+
+def create_raw_vdisk_from_template(template_folder, image_name, vpool_name, disk_name):
+    _, __ = general.execute_command('qemu-img convert -O raw {0}{1} /mnt/{2}/{3}.raw'.format(template_folder, image_name, vpool_name, disk_name))
+
+
+def create_machine_from_existing_raw_disk(machine_name, vpool_name, disk_name):
+    _, __ = general.execute_command('virt-install --connect qemu:///system -n {0} -r 512 --disk /mnt/{1}/{2}.raw,'
+                                    'device=disk --noautoconsole --graphics vnc,listen=0.0.0.0 --vcpus=1 --network network=default,mac=RANDOM,'
+                                    'model=e1000 --import'.format(machine_name, vpool_name, disk_name))
+
+
+def remove_machine_by_name(vmachine_name):
+    _, __ = general.execute_command('virsh destroy {0}'.format(vmachine_name))
+    _, __ = general.execute_command('virsh undefine {0}'.format(vmachine_name))
+
+
+def vms_with_fio_test():
+    """
+    {0}
+    """.format(general.get_function_name())
+
+    general.check_prereqs(testcase_number=1,
+                          tests_to_run=testsToRun)
+
+    api = Connection.get_connection()
+    vpool_list = api.get_component_by_name('vpools', VPOOL_NAME)
+    assert len(vpool_list), "No vpool found where one was expected"
+    vpool = vpool_list[0]
+    for disk_number in range(NUMBER_OF_DISKS):
+        create_raw_vdisk_from_template(template_target_folder, template_image, vpool['name'], "disk-{0}".format(disk_number))
+
+    assert len(vpool['vdisks_guids']) == NUMBER_OF_DISKS, "Only {0} out of {1} VDisks have been created".format(len(vpool['vdisks_guids']), NUMBER_OF_DISKS)
+
+    for vm_number in range(NUMBER_OF_DISKS):
+        create_machine_from_existing_raw_disk("machine-{0}".format(vm_number), vpool['name'], "machine-{0}".format(vm_number))
+
+    time.sleep(30)
+    vms = api.get_components('vmachines')
+    assert len(vms) == NUMBER_OF_DISKS, "Only {0} out of {1} VMachines have been created".format(len(vms), NUMBER_OF_DISKS)
+    time.sleep(350)
```

Comment:
can't we avoid these time.sleep parts, these are potential failures or might cause the tests to take longer than necessary?